src/annotator/config/test/index-test.js

```diff
@@ -191,16 +191,49 @@ describe('annotator.config.index', function() {
       assert.equal(configFrom(fakeWindow()).annotations, 'SOME_ANNOTATION_ID');
     });
 
-    it('ignores settings from JSON objects in the host page', function() {
-      fakeSharedSettings.jsonConfigsFrom.returns({foo: 'bar'});
+    [
+      'openLoginForm',
+      'openSidebar',
+    ].forEach(function(setting) {
+      it('still reads the ' + setting + ' setting from the JSON objects in the host page', function() {
+        var sharedSettings = {};
+        sharedSettings[setting] = 'bar';
+        fakeSharedSettings.jsonConfigsFrom.returns(sharedSettings);
+
+        assert.equal(configFrom(fakeWindow())[setting], 'bar');
+      });
 
-      assert.isUndefined(configFrom(fakeWindow()).foo);
+      it('still reads the ' + setting + ' setting from the hypothesisConfig() function in the host page', function() {
+        var configFuncSettings = {};
+        configFuncSettings[setting] = 'bar';
+        fakeConfigFuncSettingsFrom.returns(configFuncSettings);
+
+        assert.equal(configFrom(fakeWindow())[setting], 'bar');
+      });
     });
 
-    it('ignores settings from the hypothesisConfig() function in the host page', function() {
-      fakeConfigFuncSettingsFrom.returns({foo: 'bar'});
+    [
+      'showHighlights',
+      'branding',
+      'assetRoot',
+      'sidebarAppUrl',
```

Comment:
The `sidebarAppUrl` setting is written into the host page by the browser extension to make it load `app.html` from the extension bundle.